pkg/karmadactl/cmdinit/kubernetes/deploy.go

```diff
@@ -71,6 +90,7 @@ type CommandInitOption struct {
 	CertAndKeyFileData                 map[string][]byte
 	RestConfig                         *rest.Config
 	KarmadaAPIServerIP                 []net.IP
+	FlagSet                            *pflag.FlagSet
```

Comment:
FlagSet is not recommended as an option.

Reply:
@lonelyCZ  help to review again
I want to know if the user had set the kubeapiserver image manally.
three ways I have thought :
1: paas flagset to option and tell deploy.go if user had set the kubeapiserver image manally ( not recommended)
2: https://github.com/karmada-io/karmada/pull/1764/files#:~:text=cert%20and%20crds%20files%22),%22%22%2C%20%22Kubernetes%20apiserver%20image%22) set apiserver default value == "" and we use default value to judge user had set the value (code push now )
3: add  some options  in opts like apiserverset .  we set the apiserverset value by visit when flags init